app/controllers/ProjectSidewalkAPIController.scala

```diff
@@ -162,36 +199,107 @@ class ProjectSidewalkAPIController @Inject()(implicit val env: Environment[User,
     * @param lng1
     * @param lat2
     * @param lng2
+    * @param filetype
     * @return
     */
-  def getAccessScoreNeighborhoodsV2(lat1: Double, lng1: Double, lat2: Double, lng2: Double) = UserAwareAction.async { implicit request =>
+  def getAccessScoreNeighborhoodsV2(lat1: Double, lng1: Double, lat2: Double, lng2: Double, filetype: Option[String]) = UserAwareAction.async { implicit request =>
     apiLogging(request.remoteAddress, request.identity, request.toString)
-    Future.successful(Ok(getAccessScoreNeighborhoodsGeneric(lat1, lng1, lat2, lng2, version = 2, request.toString)))
+    val coordinates = Array(min(lat1, lat2), max(lat1, lat2), min(lng1, lng2), max(lng1, lng2))
+    if (filetype != None && filetype.get == "csv") {  // CSV format
+      val file = getAccessScoreNeighborhoodsCSV(version = 2, coordinates)
+      Future.successful(Ok.sendFile(content = file, onClose = () => file.delete()))
+    } else {  // GeoJSON format
+      Future.successful(Ok(getAccessScoreNeighborhoodsJson(version = 2, coordinates)))
+    }
   }
 
-  def getAccessScoreNeighborhoodsGeneric(lat1: Double, lng1: Double, lat2: Double, lng2: Double, version: Int, requestStr: String) = {
-    // Retrieve data and cluster them by location and label type.
-    val minLat = min(lat1, lat2)
-    val maxLat = max(lat1, lat2)
-    val minLng = min(lng1, lng2)
-    val maxLng = max(lng1, lng2)
-    val severity: Option[String] = None
+  /**
+    *
+    * Generic version of getAccessScoreNeighborHood, makes changes for v1 vs v2, for CSV file format only.
+    *
+    * @param version
+    * @param coordinates
+    * @return
+    */
+  def getAccessScoreNeighborhoodsCSV(version: Int, coordinates: Array[Double]) = {
+    val file = new java.io.File("access_score_neighborhoods.csv")
+    val writer = new java.io.PrintStream(file)
+    writer.print("Neighborhood Name,Region ID,Access Score,Coordinates,Coverage,Average Curb Ramp Score,")
+    writer.print("Average No Curb Ramp Score,Average Obstacle Score,Average Surface Problem Score,Curb Ramp Significance,")
+    writer.println("No Curb Ramp Significance,Obstacle Significance,Surface Problem Significance")
+    def featureCollectionCSV = {
```

Comment:
I don't understand why this is a "def", or why there is a separate code block here..?